File: leetcode/easy/Q448_FindAllNumbersDisappearedinanArray.py
from typing import List


class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        if not nums:
            return []

        max_num = len(nums)
        check = [False] * max_num

        for num in nums:
            check[num - 1] = True

        answer = []
        for i in range(max_num):
            if not check[i]:
                answer.append(i + 1)

        return answer

File: src/easy/Q448_FindAllNumbersDisappearedinanArray.java
package easy;

import java.util.ArrayList;
import java.util.List;

public class Q448_FindAllNumbersDisappearedinanArray {
    class Solution {
        public List<Integer> findDisappearedNumbers(int[] nums) {
            int length = nums.length;
            boolean[] check = new boolean[length];
            for(int n : nums){
                check[n-1] = true;
            }
            List<Integer> ans = new ArrayList<>();
            for (int i = 0; i < length; i++) {
                if(!check[i]) ans.add(i+1);
            }
            return ans;
        }
    }
}
